### PR TITLE
Omit duplicated observations from WQP data

### PR DESCRIPTION
**Pull request by lekoenig**
Addresses #114.

The code changes here include edits to the function `subset_wqp_SC_data()` (`2_process/src/filter_wqp_data.R`) that allow us to exclude duplicated observations that exist within the discrete WQP dataset (see #114 for further details). These edits exclude a decent chunk of samples from the target `p2_wqp_SC_data`:

```
> # Previously:
> p2_wqp_SC_data <- subset_wqp_SC_data(p2_filtered_wqp_data, omit_dups = FALSE)
> dim(p2_wqp_SC_data)
[1] 96884    26
> 
> # Now:
> p2_wqp_SC_data <- subset_wqp_SC_data(p2_filtered_wqp_data, omit_dups = TRUE)
> dim(p2_wqp_SC_data)
[1] 79031    26
```

In addition, this PR adds a new target `p2_wqp_SC_dups` that represents a subset of the duplicated observations that we could use to estimate observation error, for example.

### CODE REVIEW

File: _targets.R [hunk 1]
```diff
@@ -21,11 +21,11 @@ dir.create("3_visualize/out/nhdv2_attr_png/",showWarnings = FALSE)
 
 # Define columns of interest for harmonized WQP data
 wqp_vars_select <- c("MonitoringLocationIdentifier","MonitoringLocationName","LongitudeMeasure","LatitudeMeasure",
-                     "MonitoringLocationTypeName","OrganizationIdentifier","ActivityStartDate","ActivityStartTime.Time",
-                     "ActivityEndDate","CharacteristicName","param_group","param","USGSPCode","ActivityMediaName",
-                     "ResultSampleFractionText","HydrologicCondition","HydrologicEvent","resultVal2","resultUnits2",
-                     "ResultDetectionConditionText","ResultTemperatureBasisText","PrecisionValue","ResultStatusIdentifier",
-                     "final")
+                     "MonitoringLocationTypeName","OrganizationIdentifier","ActivityStartDate","ActivityStartDateTime",
+                     "ActivityStartTime.Time","ActivityStartTime.TimeZoneCode","CharacteristicName","param_group","param",
```
Comment (jds485) [inline _targets.R:25]:
```suggestion
                     "ActivityStartTime.Time","ActivityStartTime.TimeZoneCode","ActivityEndDate","CharacteristicName","param_group","param",
```

Comment (jds485) [reply _targets.R:25]: Did you mean to drop the end date?

**Review by jds485 — APPROVED** [review]
This looks good and runs for me. I have 2 small suggestions, and you can merge whether or not you accept them.

File: _targets.R (hunk 1 above)
Comment (lekoenig) [reply _targets.R:25]: I _did_ intentionally drop the end date here, although I'm also happy to keep it. `ActivityEndDate` is `NA` for the vast majority of discrete samples. I had originally kept this column for QC purposes (_does end date equal start date?_), but for these sites it's hard to interpret the meaning if the end date is different, and so we're not really using it. From what I could tell the temperature [repo](https://github.com/USGS-R/2wp-temp-observations/blob/main/1_wqp_pull/src/get_wqp_data.R) doesn't retain that column, so I omitted it here.

File: 2_process/src/filter_wqp_data.R [hunk 2]
```diff
@@ -50,14 +52,74 @@ subset_wqp_SC_data <- function(filtered_data){
...
+    arrange(match(param, c("Specific conductance, lab",
+                           "Specific conductance",
+                           "Specific conductance, field",
+                           "Specific conductance, field, mean"))) %>%
+    slice(1) %>%
+    ungroup()
+  
+  if(omit_dups == "TRUE"){
+    return(SC_data_subset_omit_dups)
+  } else {
```
[the remark this answers is not in the document]
Comment (lekoenig) [reply 2_process/src/filter_wqp_data.R:84]: Great idea - thanks!

File: _targets.R (hunk 1 above)
Comment (jds485) [reply _targets.R:25]: okay, dropping `ActivityEndDate` sounds good!

**Pull request merged**